### ISSUE TITLE
I only have the training set image. How do I set the path to train, val and test in the yaml file of the data set? Is there a way to skip each epoch of validation?

### ISSUE THREAD

**Issue by yangershuai627** [ultralytics/ultralytics#18549]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I only have the training set image. How do I set the path to train, val and test in the yaml file of the data set? Is there a way to skip each epoch of validation?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yangershuai627, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Specifically, for your case, when configuring your dataset in the `.yaml` file, the paths such as `train`, `val`, and `test` can be customized to match the dataset structure you have. To skip validation, consider configuring the `val` path to point to an empty folder, or see if the `--noval` flag suits your use case.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

⚡ This is an automated response, but please rest assured that an Ultralytics engineer will assist you further as soon as possible!

**Comment by Skillnoob** [comment]
Add `val=False` to `.train(...)`

**Comment by yangershuai627** [comment]
> Add `val=False` to `.train(...)`

By setting val=False, I successfully skipped the verification phase of each epoch. I notice that I need to point the path to the validation set in the configuration file to the location where the training set image is stored.

**Comment by yangershuai627** [comment]
> @yangershuai627 How can you be sure then, your model is performing good enough, if it hasn't been validated?
> 
> skip validation every epoch stands above.

I did not consider this problem, my original intention was to combine the train set and test set images into trainval set as the train set training model.

**Comment by Skillnoob** [comment]
You can modify your yaml like this if you want to combine the test and train set:
```
path: ...
train: 
  - images/train
  - images/test
val: images/val
test: ...

names:
    ...
```
I would not reccomend to turn off validation, as then you won't know how the model performs.

**Comment by Skillnoob** [comment]
You always need val data, Ultralytics requires you to have val data.
